Core/clim-core/commands/commands.lisp

```diff
@@ -673,50 +673,47 @@ examine the type of the command menu item to see if it is
-                                      &key gesture &allow-other-keys)
-                     arg
-                   (declare (ignore name))
-                   (let ((command-args (loop for a in args
-                                             for i from 0
-                                             if (eql i arg-index)
-                                               collect 'object
-                                             else
-                                               collect (getf (cddr a) :default)
-                                             end))
-                         (translator-name (intern (format nil
-                                                          ".~A-ARG~D."
-                                                          command-name
-                                                          arg-index)
-                                                  (symbol-package command-name))))
-                     (multiple-value-bind (gesture translator-options)
-                         (if (listp gesture)
-                             (values (car gesture) (cdr gesture))
-                             (values gesture nil))
-                       (destructuring-bind (&key (documentation
-                                                  `((object stream)
-                                                    (orf stream *standard-output*)
-                                                    (format stream "~A "
-                                                            ,readable-command-name)
-                                                    (present object (presentation-type-of object) ; type?
-                                                             :stream stream
-                                                             :acceptably nil
-                                                             :sensitive nil))
-                                                  documentationp)
-                                            &allow-other-keys)
-                           translator-options
-                       `(define-presentation-to-command-translator
-                            ,translator-name
-                            (,(eval ptype) ,command-name ,command-table
-                             :gesture ,gesture
-                             ,@(unless documentationp `(:documentation ,documentation))
-                             ,@translator-options)
-                          (object)
-                          (list ,@command-args)))))))))
+  (let* ((readable-command-name
+           (command-name-from-symbol command-name)) ; XXX or :NAME
+         (default-documentation
+           `((object stream)
+             (orf stream *standard-output*)
+             (format stream "~A " ,readable-command-name)
+             (present object (presentation-type-of object) ; type?
+                      :stream stream
+                      :acceptably nil
+                      :sensitive nil))))
```

Comment:
Could move this into the `flet` as a function to communicate the fact that it is not always needed. Also reduces consing a bit in some case, but I don't know whether we expand enough commands for that to matter.

Reply:
fixed